noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_init.nr

```diff
@@ -23,222 +24,106 @@ struct PrivateKernelInitCircuitPrivateInputs {
         let mut builder = PrivateKernelInitInputsBuilder::new();
 
-        // Logs for the private call.
-        let encrypted_logs_hashes = [16, 36];
-        let encrypted_log_preimages_length = [100, 75];
-        let unencrypted_logs_hashes = [26, 46];
-        let unencrypted_log_preimages_length = [50, 25];
-        builder.private_call.public_inputs.add_encrypted_log(encrypted_logs_hashes[0], encrypted_log_preimages_length[0]);
-        builder.private_call.public_inputs.add_unencrypted_log(
-            unencrypted_logs_hashes[0],
-            unencrypted_log_preimages_length[0]
-        );
-        builder.private_call.public_inputs.add_encrypted_log(encrypted_logs_hashes[1], encrypted_log_preimages_length[1]);
-        builder.private_call.public_inputs.add_unencrypted_log(
-            unencrypted_logs_hashes[1],
-            unencrypted_log_preimages_length[1]
-        );
+        // note_hash_read_requests
+        builder.private_call.append_note_hash_read_requests(2);
+        let note_hash_read_requests = builder.private_call.note_hash_read_requests.storage;
 
-        let expected_unencrypted_logs = [
-            builder.private_call.public_inputs.unencrypted_logs_hashes.storage[0], builder.private_call.public_inputs.unencrypted_logs_hashes.storage[1]
-        ];
-        let expected_encrypted_logs = [
-            builder.private_call.public_inputs.encrypted_logs_hashes.storage[0], builder.private_call.public_inputs.encrypted_logs_hashes.storage[1]
-        ];
+        // encrypted_logs_hashes
+        builder.private_call.append_encrypted_log_hashes(1);
+        let encrypted_log_hashes = builder.private_call.encrypted_logs_hashes.storage;
 
         let public_inputs = builder.execute();
-
-        // Check the first nullifier is hash of the signed tx request
-        let tx_hash = builder.tx_request.hash();
-        assert_eq(public_inputs.end.new_nullifiers[0].value(), tx_hash);
-        assert(
-            array_eq(
-                public_inputs.end.encrypted_logs_hashes.map(|l: ScopedEncryptedLogHash| l.log_hash),
-                expected_encrypted_logs
-            )
+        assert_array_eq(
+            public_inputs.validation_requests.note_hash_read_requests,
+            [note_hash_read_requests[0], note_hash_read_requests[1]]
         );
-        assert(
-            array_eq(
-                public_inputs.end.unencrypted_logs_hashes.map(|l: ScopedLogHash| l.log_hash),
-                expected_unencrypted_logs
-            )
+        assert_array_eq(
+            public_inputs.end.encrypted_logs_hashes,
+            [encrypted_log_hashes[0]]
         );
     }
-
-    #[test]
-    fn default_max_block_number() {
-        let mut builder = PrivateKernelInitInputsBuilder::new();
-        let public_inputs = builder.execute();
-
-        assert(public_inputs.validation_requests.for_rollup.max_block_number.is_none());
-    }
-
-    #[test]
-    fn propagate_max_block_number_request() {
```

Comment:
Tests for propagating data are done in the tests for `PrivateKernelCircuitPublicInputsComposer`.